Add Error Boundary to useTransition API docs (#6354)
reactjs/react.dev: src/content/reference/react/useTransition.md

---

### Displaying an error to users with a error boundary {/*displaying-an-error-to-users-with-error-boundary*/}

<Canary>

Error Boundary for useTransition is currently only available in React's canary and experimental channels. Learn more about [React's release channels here](/community/versioning-policy#all-release-channels).

</Canary>

> If you'd like to display an error to your users when using `useTransition`, you can use an [error boundary](/reference/react/Component#catching-rendering-errors-with-an-error-boundary). To use an error boundary, wrap the component where you are calling the `useTransition` in an error boundary. If the function passed to `startTransition` is rejected, the fallback for the error boundary will be displayed.

Reviewer: Nit/Suggestion: Rephrasing to lead with why you'd want to consider an error boundary

```suggestion
If a function passed to `startTransition` throws an error, you can display an error to your user with an [error boundary](/reference/react/Component#catching-rendering-errors-with-an-error-boundary). To use an error boundary, wrap the component where you are calling the `useTransition` in an error boundary. Once the function passed to `startTransition` errors, the fallback for the error boundary will be displayed.
```

<Sandpack>

```js AddCommentContainer.js active
import { useTransition } from "react";
import { ErrorBoundary } from "react-error-boundary";

export function AddCommentContainer() {
  return (
    <ErrorBoundary fallback={<p>⚠️Something went wrong</p>}>
        <AddCommentButton />
    </ErrorBoundary>
  );
}

> function addComment(comment) {

Reviewer: We can probably remove the comment param and null check and just always throw?

Author: True the check isn't necessary, but I kind of wanted to make it feel more realistic when we'd throw an error

---

### Displaying an error to users with a error boundary {/*displaying-an-error-to-users-with-error-boundary*/}

<Canary>

Error Boundary for useTransition is currently only available in React's canary and experimental channels. Learn more about [React's release channels here](/community/versioning-policy#all-release-channels).

</Canary>

> If you'd like to display an error to your users when using `useTransition`, you can use an [error boundary](/reference/react/Component#catching-rendering-errors-with-an-error-boundary). To use an error boundary, wrap the component where you are calling the `useTransition` in an error boundary. If the function in `startTransition` is rejected the fallback for the error boundary will be displayed.

Reviewer 2: I think "rejected" is a term typically just used with Promises. "throws" would be more general.

Reviewer 3: Technically since it's an async function it is a promise that rejects after the first `await`, right? Not sure the clearest way to say this.

Reviewer 2: "throws" feels right to me for async functions as you can `catch` the error around the `await`. Just a feeling though!

<Sandpack>

```js SubmitContainer.js active

> 

Reviewer: ```suggestion
```

import { useTransition } from "react";

> import { ErrorBoundary } from "react-error-boundary";

Reviewer 2: The React docs should probably use React's APIs directly?

Author: ErrorBoundary isn't part of a React API. See note [here](https://react.dev/reference/react/Component#catching-rendering-errors-with-an-error-boundary) and package [here](https://www.npmjs.com/package/react-error-boundary)

export function SubmitContainer() {
  return (
    <ErrorBoundary fallback={<p>⚠️Something went wrong</p>}>
        <SubmitButton />
    </ErrorBoundary>
  );
}

function SubmitButton() {
  const [pending, startTransition] = useTransition();

  return (
    <button 
      disabled={pending} 
      onClick={() => {
        startTransition(() => {
          throw new Error();
        });
      }}>

>         Submit

Reviewer: Maybe we could make this button label be like: "Trigger Error Boundary" to be clearer

Reviewer: Or, lean into a real-world example where we'd use a `useTransition` like the example of loading indicators: https://react-nzxyyeyvn-fbopensource.vercel.app/reference/react/useTransition#preventing-unwanted-loading-indicators

Author: I'll update this example to be more product specific based on yours and @sophiebits's suggestions

      </button>      
  );
}
```

```js App.js hidden
import { SubmitContainer } from "./SubmitContainer.js";

export default function App() {
  return <SubmitContainer />;
}
```

```js index.js hidden
// TODO: update to import from stable

> // react instead of canary once the `use`

Reviewer 2: copypasta? Did you mean `useTransition`?
